process/smartContract/scQueryService.go

```diff
@@ -101,6 +115,74 @@ func (service *SCQueryService) checkVMOutput(vmOutput *vmcommon.VMOutput) error
 	return nil
 }
 
+// ComputeTransactionCost will estimate how many gas a transaction will consume
+func (service *SCQueryService) ComputeTransactionCost(tx *transaction.Transaction) (*big.Int, error) {
+	txType, err := service.txTypeHandler.ComputeTransactionType(tx)
+	if err != nil {
+		return nil, err
+	}
+
+	tx.GasPrice = 1
+
+	switch txType {
+	case process.MoveBalance:
+		return service.estimateMoveBalance(tx)
+	case process.SCDeployment:
+		return service.estimateSCDeployment(tx)
+	case process.SCInvoking:
+		return service.estimateSCInvoking(tx)
+	default:
+		return nil, process.ErrWrongTransaction
+	}
+}
+
+func (service *SCQueryService) estimateMoveBalance(tx *transaction.Transaction) (*big.Int, error) {
+	cost := service.economicsFee.ComputeFee(tx)
```

Comment:
these estimations should be in a class on top of scqueryservice. scquery only resolves the execution querry.

Reply:
done.